source/extensions/filters/http/istio_stats/istio_stats.cc

```diff
@@ -574,42 +516,68 @@ struct Config : public Logger::Loggable<Logger::Id::filter> {
     }
   }
 
-  void addCounter(Stats::StatName metric, const Stats::StatNameTagVector& tags,
-                  const StreamInfo::StreamInfo& info,
-                  Stats::StatNameDynamicPool& pool, uint64_t amount = 1) {
-    if (metric_overrides_) {
-      if (metric_overrides_->drop_.contains(metric)) {
+  // RAII for stream context propagation.
+  struct StreamOverrides {
+    StreamOverrides(Config& parent, const StreamInfo::StreamInfo& info,
+                    const Http::RequestHeaderMap* request_headers = nullptr,
+                    const Http::ResponseHeaderMap* response_headers = nullptr,
+                    const Http::ResponseTrailerMap* response_trailers = nullptr)
+        : parent_(parent) {
+      if (parent_.metric_overrides_) {
+        parent_.metric_overrides_->initialize(
```

Comment:
This is buggy, need 7424dcd0b7931cfc550776ed2824a50e61fba50c as well.

Reply:
I think there's some other changes missing.. there were some conflicts cherry picking and tests are failing now.

Reply:
We did the same pushes :)